docs/src/pages/components/data-grid/filtering/QuickFilteringGrid.js

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import TextField from '@material-ui/core/TextField';
+import {
+  DataGrid,
+  GridToolbarColumnsButton,
+  GridToolbarContainer,
+  GridToolbarDensitySelector,
+  GridToolbarFilterButton,
+} from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import ClearIcon from '@material-ui/icons/Clear';
+import SearchIcon from '@material-ui/icons/Search';
+import { createMuiTheme } from '@material-ui/core/styles';
+import { makeStyles } from '@material-ui/styles';
+
+const defaultTheme = createMuiTheme();
+const useStyles = makeStyles(
+  (theme) => ({
+    toolbar: {
+      '&&': {
```

Comment:
This is hiding something
```suggestion
```

Reply:
I have found what it was hiding. The styles of `MuiDataGrid-toolbarContainer` have an extra level of specificity. We don't need to have it. Since it can be composed, I propose we flatten the styles of `GridToolbarContainer` and similar components.

Reply:
I removed it

Reply:
> The styles of MuiDataGrid-toolbarContainer have an extra level of specificity. We don't need to have it.

The same for `box-sizing: border-box`. We apply it on everything but `CssBaseline` already sets it.

Reply:
@m4theushw CssBaseline is not guaranteed to be present. It's not straightforward to introduce in an existing codebase. For new project, we do encourage its usage.

Reply:
@oliviertassinari We can keep `box-sizing: border-box` but applying only on those elements that really need it. The ".MuiDataGrid-root *" selector is so broad that it overwrites the new value set by the TextField.

Reply:
I have opened #2177, a good one for the community.

Reply:
> We can keep box-sizing: border-box but applying only on those elements that really need it.

It's what I did in the core, I won't object 😁

> The ".MuiDataGrid-root *" selector is so broad that it overwrites the new value set by the TextField.

Right, we can even argue that it can create bugs for developers with custom render cells.